libs/symphonia/src/bin/symphtest.rs

```diff
@@ -1,65 +1,197 @@
+struct Params {
+    latency: NetworkLatencyParams,
+    participants: ParticipantParams,
 }
 
 fn main() {
+    env_logger::from_env(Env::default().default_filter_or("symphonia=trace,warn")).init();
     let opt: Opt = Opt::from_args();
-    println!("{:?}", opt);
     smol::block_on(async move {
-        env_logger::from_env(Env::default().default_filter_or("symphonia=trace,warn")).init();
-        let mut harness = Harness::new(MockNet {
-            latency_mean_ms: opt.latency_mean_ms,
-            latency_variance: opt.latency_variance,
-            loss_prob: opt.loss_prob,
-        });
-        for particpant_weight in opt.participant_weights.iter() {
-            harness =
-                harness.add_participant(tmelcrypt::ed25519_keygen().1, particpant_weight.clone());
+        match opt {
+            Opt::Rounds {
+                rounds,
+                latency_mean_ms,
+                latency_standard_deviation,
+                loss_prob,
+                participant_weights,
+            } => {
+                for _ in 0..rounds {
+                    let mock_net = MockNet {
+                        latency_mean_ms,
+                        latency_standard_deviation,
+                        loss_prob,
+                    };
+                    // TODO: avoid clone by using immutable vector conversion before loop
+                    run_round(participant_weights.clone(), mock_net).await
+                }
+            }
+            Opt::Parameterize {
+                rounds,
+                file_name,
+                test_count,
+            } => {
+                // Load file and deserialize into params
+                let mut path = env::current_dir().expect("Failed to get current directory");
+                path.push(file_name);
+                let file_contents = fs::read_to_string(path).expect("Unable to read file");
+                let params: Params =
+                    toml::from_str(&file_contents).expect("Unable to deserialize params");
+
+                // Run test cases
+                for _ in 0..test_count {
+                    // Sample latency and create mock network
+                    let (latency_mean_ms, latency_standard_deviation, loss_prob) =
+                        params.latency.sample();
+                    let mock_net = MockNet {
+                        latency_mean_ms,
+                        loss_prob,
+                        latency_standard_deviation,
+                    };
+
+                    // Sample participants and run rounds
+                    let participant_weights = params.participants.sample();
+                    for _ in 0..rounds {
+                        run_round(participant_weights.clone(), mock_net.clone()).await
+                    }
+                }
+            }
         }
-        harness.run().await
     });
 }
+
+async fn run_round(participant_weights: Vec<u64>, mock_net: MockNet) {
+    let mut harness = Harness::new(mock_net);
```

Comment:
both subcommands use this, I kept the old subcommand in case we want to manually test one scenario.  The parameterize is better for a search oriented thing which we can leave running.

Reply:
Sure